dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/TimeoutFilter.java

```diff
@@ -36,21 +37,40 @@
 
     private static final Logger logger = LoggerFactory.getLogger(TimeoutFilter.class);
 
+    private static final String TIMEOUT_FILTER_START_TIME = "timeout_filter_start_time";
+
     @Override
     public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
-        long start = System.currentTimeMillis();

```

Comment:
`System.currentTimeMillis();` is not good choice for interval measurement.
http://sangsoonam.github.io/2017/03/01/do-not-use-curenttimemillis-for-time-interval.html
https://github.com/openshift/jenkins-plugin/issues/76
http://pzemtsov.github.io/2017/07/23/the-slow-currenttimemillis.html

Reply:
I know.
I have read an article and it seems that it is better to start another thread to get time regularly? But I think this is not a bottleneck right now.

Reply:
Let's keep it easy, functionality then performance.